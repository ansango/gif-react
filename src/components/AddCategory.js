import { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputChange = (event) => setInputValue(event.target.value);
  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length > 0) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };
  return (
    <div className="py-12 mt-8 max-w-md">
      <form onSubmit={handleSubmit} className="grid grid-cols-1 gap-6">
        <label className="block">
          <span>Search a GIF</span>
          <input
            type="text"
            value={inputValue}
            onChange={handleInputChange}
            className="mt-1 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
          />
        </label>
        <button type="submit">Search</button>
      </form>
    </div>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
