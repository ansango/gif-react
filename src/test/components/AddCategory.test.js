import AddCategory from "../../components/AddCategory";
const { shallow } = require("enzyme");

describe("Test <AddCategory/>", () => {
  const setCategories = () => {};

  test("should render component", () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    expect(wrapper).toMatchSnapshot();
  });
});
