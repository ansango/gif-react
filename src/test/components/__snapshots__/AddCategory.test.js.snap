// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory/> should render component 1`] = `
<div
  className="py-5 max-w-xl mx-auto"
>
  <form
    className="md:flex items-center"
    onSubmit={[Function]}
  >
    <label
      className="md:w-2/3"
    >
      <input
        className="w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 p-3"
        onChange={[Function]}
        placeholder="Search a GIF"
        type="text"
        value=""
      />
    </label>
    <div
      className="md:w-1/3 md:pl-5"
    >
      <button
        className="bg-blue-400 text-white p-3 rounded-lg w-full my-3"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
